File: Gato.java
/*
                  Universidad de las fuerzas armadas
                   Tecnologías de la información
Nombre: Juan Pablo Cano                             NRC: 13930
Fecha: 13-11-2023
                            
Autor: Juan Pablo Cano 
 */
package gato_perro;

public class Gato {
    String nombre;
    int edad;
    String raza;

    public Gato(String nombre, int edad, String raza) {
        this.nombre = nombre;
        this.edad = edad;
        this.raza = raza;
    }
    
    public void maullar (){
        System.out.println(this.nombre+" empezo a maullar");
    }
    
    public void comer (){
        System.out.println(this.nombre+" empezo a comer");
        
    }
    
    public void saltar (){
        System.out.println(this.nombre+" empezo a saltar");
    }
    
    public void correr (){
        System.out.println(this.nombre+" empezo a correr");
    }
    
}

File: C++/Gato.cpp
#include <iostream>
#include <string>

class Gato {
private:
    std::string nombre;
    int edad;
    std::string raza;

public:
    Gato(std::string nombre, int edad, std::string raza) : nombre(nombre), edad(edad), raza(raza) {}

    void maullar() {
        std::cout << nombre << " empezo a maullar" << std::endl;
    }

    void comer() {
        std::cout << nombre << " empezo a comer" << std::endl;
    }

    void saltar() {
        std::cout << nombre << " empezo a saltar" << std::endl;
    }

    void correr() {
        std::cout << nombre << " empezo a correr" << std::endl;
    }
};

int main() {
    // Ejemplo de uso
    Gato miGato("Pelusa", 3, "Persa");
    miGato.maullar();
    miGato.comer();
    miGato.saltar();
    miGato.correr();

    return 0;
}
